src/java.desktop/share/classes/sun/awt/image/GifImageDecoder.java

```diff
@@ -424,6 +433,9 @@ private int sendPixels(int x, int y, int width, int height,
                             }
                         }
                         runstart = -1;
+                        if (replaceTransPixelsInSavedImage) {
```

Comment:
We are hitting this check for each transparent pixel for all frames with transparent pixel index > 0 and not initial frame until saved_image is not null. In the .gif present in this PR we are checking this condition for each transparent pixel of 2nd and 3rd frame. If GIF contains many frames and disposal method save comes in later part of animation we will hit this check many times.

I think its better to just initialize saved_image with proper transparent pixel index when it is getting created. Then we can completely remove state management of `replaceTransPixelsInSavedImage` and this check.

Reply:
OK; I think this thread is moot having addressed https://github.com/openjdk/jdk/pull/25044#issuecomment-2912699099 .